### ISSUE TITLE
pattern-not can negate a formula

### ISSUE THREAD

**Issue by emjin** [semgrep/semgrep#5854]
**Is your feature request related to a problem? Please describe.**
Pattern-not can only have one pattern, not a formula

**Describe the solution you'd like**
Pattern-not takes a formula

**Describe alternatives you've considered**
Rewrite rule

**Use case**
Make it less annoying to negate a rule

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-1723/pattern-not-can-negate-a-formula. _Note_: this link is for r2c use only and is not accessible publicly.

**Comment by IagoAbal** [comment]
Can we do this with the rule schema used by CLI? Or do we need a `pattern-neg`?

**Comment by kopecs** [comment]
This is possible with the rule schema; it would be something like:

```yml
  pattern-not:
    type: object
    properties:
      pattern-not:
        title: Do not return finding where Semgrep pattern matches exactly
        oneOf:
          - type: string
          - anyOf: <similar to https://github.com/returntocorp/semgrep-interfaces/blob/develop/rule_schema.yaml#L5-L23>
    required:
      - pattern-not
```

**Comment by IagoAbal** [comment]
Related to #5840

**Comment by IagoAbal** [comment]
We should try doing this, it's probably cheap nowadays.

**Comment by IagoAbal** [comment]
Closed by #6474
